public/pages/workflow_detail/agentic_search/test_flow/index_selector.tsx

```diff
@@ -55,20 +58,28 @@ export function IndexSelector(props: IndexSelectorProps) {
   const [indexOptions, setIndexOptions] = useState<
     { value: string; text: string }[]
   >([]);
+
+  // Optionally add an "ALL INDICES" option for eligible agent types (conversational)
   useEffect(() => {
-    setIndexOptions([
-      {
-        text: ALL_INDICES,
-        value: '',
-      },
+    let eligibleIndexOptions = [
       ...Object.values(indices || {})
         .filter((index) => !index.name.startsWith('.')) // Filter out system indices
         .map((index) => ({
           value: index.name,
           text: index.name,
         })),
-    ]);
-  }, [indices]);
+    ];
+    if (props.agentType === AGENT_TYPE.CONVERSATIONAL) {
+      eligibleIndexOptions = [
+        {
+          text: ALL_INDICES,
+          value: '',
+        },
+        ...eligibleIndexOptions,
+      ];
+    }
+    setIndexOptions(eligibleIndexOptions);
+  }, [indices, props.agentType]);
```

Comment:
Do we want to have a test for this change?

Reply:
Good idea, added in latest commit.